### PR TITLE
Print TE CUDA kernel

### PR DESCRIPTION
**Pull request by bertmaher**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42692 Print TE CUDA kernel**

Differential Revision: [D22986112](https://our.internmc.facebook.com/intern/diff/D22986112)

### CODE REVIEW

File: torch/csrc/jit/tensorexpr/cuda_codegen.cpp [hunk 1]
```diff
@@ -718,6 +717,18 @@ class NoThreadIdxRewriter : public IRMutator {
   bool need_rewrite_ = false;
 };
 
+std::ostream& operator<<(
```
Comment (zheng-xq) [inline torch/csrc/jit/tensorexpr/cuda_codegen.cpp:720]: I am not a fan of overloading "operator<<" for a generic data structure. I am perfectly fine if it is either static or within an anonymous namespace.

File: benchmarks/fastrnns/bench.py [hunk 2]
```diff
@@ -73,11 +74,15 @@ def train_batch(modeldef):
         gc.collect()
 
         fwd_start_event.record()
+        #print(modeldef.forward.graph_for(*modeldef.inputs))
         forward_output = modeldef.forward(*modeldef.inputs)
         fwd_end_event.record()
 
         # XXX: Use if need to print something
-        # print(modeldef.forward.graph_for(*modeldef.inputs))
+        global print_once
```
Comment (zheng-xq) [inline benchmarks/fastrnns/bench.py:82]: If I remember correctly, this function is called for different rnn_creator, so why a global "print_once"? I am perfectly fine if this flag is local to a particular training, and only once across all the training steps.

**Review by zheng-xq — APPROVED** [review]

File: torch/csrc/jit/tensorexpr/cuda_codegen.cpp [hunk 3]
```diff
@@ -963,11 +957,6 @@ void CudaCodeGen::CompileToNVRTC(
   int major, minor;
   getMajorMinor(prop, major, minor);
 
-#if DEBUG_PRINT
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/cuda_codegen.cpp:966]: Do we not need this in debug logs?

File: benchmarks/fastrnns/bench.py [hunk 4]
```diff
@@ -55,6 +55,7 @@ def elapsed_time(self, end_event):
         return end_event.time - self.time
 
 
+print_once = True
```
Comment (ZolotukhinM) [inline benchmarks/fastrnns/bench.py:58]: Changes to this file seem unrelated. Are they intentional?

File: benchmarks/fastrnns/bench.py (hunk 2 above)
Comment (bertmaher) [reply benchmarks/fastrnns/bench.py:82]: Oops, this is not supposed to be here :)

File: torch/csrc/jit/tensorexpr/cuda_codegen.cpp (hunk 1 above)
Comment (bertmaher) [reply torch/csrc/jit/tensorexpr/cuda_codegen.cpp:720]: I'll make this static, but I'm not sure I understand the objection?

File: torch/csrc/jit/tensorexpr/cuda_codegen.cpp (hunk 3 above)
Comment (bertmaher) [reply torch/csrc/jit/tensorexpr/cuda_codegen.cpp:966]: Seems kind of low-value to me, unless someone really wants it I was inclined to remove it for the sake of spamming the console less.
